### PR TITLE
feat(instrumentation-document-load): performance paint timing events

### PR DESCRIPTION
**Pull request by kkruk-sumo**
## Which problem is this PR solving?

Added new events to the root span `documentLoad`:
- `firstPaint`
- `firstContentfulPaint`

These events can be used in RUM and to better understand website's loading performance.

## Short description of the changes

The `_getEntries` method has been moved to utils as it doesn't need to have class instance. New util `addSpanPerformancePaintEvents` enhances given span with the collected paint events.

### CODE REVIEW

File: plugins/web/opentelemetry-instrumentation-document-load/src/utils.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        }
+      });
+    }
+  }
+
+  return entries;
+};
+
+const performancePaintNames = {
+  'first-paint': EventNames.FIRST_PAINT,
```
Comment (johnbley) [inline plugins/web/opentelemetry-instrumentation-document-load/src/utils.ts:64]: I don't see a lot of dashes in names/semantic conventions in otel... `.` and `_` yes, but not `-`.  Perhaps the camel case names are fine as they are?  Most other browser `performance` event names are passed through unchanged.

**Review by johnbley — COMMENTED** [review]
Generally looks good, but I think the event names can be improved.

File: plugins/web/opentelemetry-instrumentation-document-load/src/utils.ts (hunk 1 above)
Comment (kkruk-sumo) [reply plugins/web/opentelemetry-instrumentation-document-load/src/utils.ts:64]: So API defines them as `first-paint` and I proposed to use `firstPaint`.
The `EventNames` enum is for our OT enum names and here I define how to map them.
Other events are camel cased and I wanted to follow this convention.

Comment (johnbley) [reply plugins/web/opentelemetry-instrumentation-document-load/src/utils.ts:64]: Oh, I was reading it backwards!  Apologies!

Comment (obecny) [reply plugins/web/opentelemetry-instrumentation-document-load/src/utils.ts:64]: @johnbley does it mean you are ok with this changes then ?

**Comment by obecny** [comment]
@kkruk-sumo can you please fix the tests, and perhaps allow maintainers to allow changes so this way it will be easier to merge when it is outdated

**Comment by kkruk-sumo** [comment]
@obecny Sorry. Firstly I had issues with eslint (finally found out that I had the wrong eslint version) and then `main` was failing. Now it passes.

**Comment by mhennoch** [comment]
LGTM

File: plugins/web/opentelemetry-instrumentation-document-load/src/utils.ts (hunk 1 above)
Comment (johnbley) [reply plugins/web/opentelemetry-instrumentation-document-load/src/utils.ts:64]: Yes, it looks good.  Sorry I didn't officially, approve, will do so.

**Review by johnbley — APPROVED** [review]

**Review by obecny — APPROVED** [review]
lgtm

**Comment by kkruk-sumo** [comment]
@obecny Can we merge this PR?

**Comment by obecny** [comment]
@kkruk-sumo yes, but it cannot be merged until it is rebased with base branch . So please merge with latest or allow editing by maintainers :) thx
https://github.com/open-telemetry/opentelemetry-js-contrib/pull/484#issuecomment-848305163

**Comment by kkruk-sumo** [comment]
@obecny Apologies. I didn't know the PR needs to be up to date. Unfortunately I don't have permissions to allow maintainers to edit this PR or at least I don't see such option. 
Branch should be up to main now :)

**Comment by obecny** [comment]
> @obecny Apologies. I didn't know the PR needs to be up to date. Unfortunately I don't have permissions to allow maintainers to edit this PR or at least I don't see such option.
> Branch should be up to main now :)

merged :), thx for contribution

**Pull request merged**